contracts/modules/capital/SafeTracker.sol

```diff
@@ -0,0 +1,159 @@
+    _;
+  }
+
+  modifier onlySwapOperator {
+    require(msg.sender == internalContracts[uint(ID.P1)], "SafeTracker: Not SwapOperator");
+    _;
+  }
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor(address _safe, address _usdc, address _aweth, address _debtUsdc) {
+    safe = _safe;
+    usdc = IERC20(_usdc);
+    aweth = IERC20(_aweth);
+    debtUsdc = IERC20(_debtUsdc);
+  }
+
+  /**
+  * @dev Gets the balance of the safe
+  * @return An uint256 representing the amount of the safe.
+  */
+  function totalSupply() external view onlyPool returns (uint256) {
+    return _calculateBalance();
+  }
+
+  /**
+  * @dev Gets the balance of the safe
+  * @return An uint256 representing the amount of the safe.
+  */
+  function balanceOf(address) external view onlyPool returns (uint256) {
+    return _calculateBalance();
+  }
+
+  /**
+  * @dev Updates invested USDC
+  */
+  function updateInvestedUSDC(uint _investedUSDC) external {
+    require(msg.sender == safe, "SafeTracker: not safe");
+    investedUSDC = _investedUSDC;
+  }
+
+  /**
+  * @dev emits Transfer event only if it's called by Pool or SwapOperator
+  */
+  function transfer(address, uint256) external onlyPoolAndSwapOperator returns (bool){
+    emit Transfer(address(0), address(0), 0); // add arguments
+    return true;
+  }
+
+  /**
+  * @dev emits Transfer event only if it's called by Pool or SwapOperator
+  */
+  function transferFrom(address, address, uint256) external override onlyPoolAndSwapOperator returns(bool) {
+    emit Transfer(address(0), address(0), 0); // add arguments
+    return true;
+  }
+
+  function allowance(address, address) external pure returns (uint256){
+    return 0;
+  }
+
+  function approve(address, uint256) external view override onlyPoolAndSwapOperator returns (bool){
+    return true;
+  }
+
+  /**
+  * @dev Fetches all necessary information about the tokens that are used in the safe and calculates the balance
+  * @return balance ETH value of the safe.
+  */
+  function _calculateBalance() internal view returns (uint256 balance) {
+
+    // eth in the safe and aweth balance, aweth is 1:1 to eth
+    uint ethAmount = address(safe).balance + aweth.balanceOf(safe);
+
+    IPriceFeedOracle priceFeedOracle = pool().priceFeedOracle();
+
+    // usdc actually in the safe
+    uint usdcAmount = usdc.balanceOf(safe) + investedUSDC;
+    uint usdcInEth = priceFeedOracle.getEthForAsset(address(usdc), usdcAmount);
+
+    // usdc debt (borrowed usdc)
+    uint usdcDebtAmount = debtUsdc.balanceOf(safe);
+    uint usdcDebtInEth = priceFeedOracle.getEthForAsset(address(usdc), usdcDebtAmount);
+
+    return ethAmount + usdcInEth - usdcDebtInEth;
+  }
+
+  function requestTransfer(uint amount) external onlySafe {
+    transferAmount = amount;
+  }
+
+  function executeTransfer() external onlySwapOperator {
+    transferAmount = 0;
+  }
+
+  function getSafeAddress() external view returns (address) {
```

Comment:
Seems a bit redundant as safe is already public. any specific reason why you want this here?

Reply:
Cleaned it up already